### PR TITLE
add hourly support for regrid_time

### PR DESCRIPTION
**Pull request by JaroCamphuijsen**
This pull request adds support for 1hr, 3hr and 6hr frequencies in the regrid_time method of the preprocessor.

### CODE REVIEW

**Comment by mattiarighi** [comment]
It would be good to update the documentation accordingly.

**Comment by JaroCamphuijsen** [comment]
I suppose I should also add tests for 3hr and 6hr data

**Comment by mattiarighi** [comment]
This affects @ledm's code, if it's fine with him, it's fine with me as well.

**Comment by ledm** [comment]
It looks to me like it just extends the current capabilities and shouldn't affect the functionality of the older time-regridding functions, right?

Can you change the header description in the function `regrid_time` to include the new times, please?

Otherwise, I'm happy with this.

File: doc/esmvalcore/preprocessor.rst [hunk 1]
```diff
@@ -304,8 +304,9 @@ See also :func:`esmvalcore.preprocessor.annual_mean`.
 This function aligns the time points of each component dataset to allow the
 subtraction of two Iris cubes from different datasets. The operation makes the
 datasets time points common and sets common calendars; it also resets the time
-bounds and auxiliary coordinates to reflect the artifically shifted time
-points. The current implementation works only for monthly and daily data.
+bounds and auxiliary coordinates to reflect the artificially shifted time
+points. The current implementation works for monthly, daily, 6 hourly, 3
+hourly and hourly data.
```
Comment (mattiarighi) [inline doc/esmvalcore/preprocessor.rst]: Please also add the arguments it takes (see other cases).

**Pull request merged**